packages/wallet/src/persistence/inMemoryStores.ts

```diff
@@ -0,0 +1,99 @@
+import { Asset, Cardano, ProtocolParametersRequiredByWallet, TimeSettings } from '@cardano-sdk/core';
+import { CollectionStore, DocumentStore, GetDocId, OrderedCollectionStore, RewardAccountDocument } from './types';
+import { Observable, of } from 'rxjs';
+import { sortedIndexBy } from 'lodash-es';
+
+export class InMemoryCollectionStore<T> implements CollectionStore<T> {
+  protected readonly docs: T[] = [];
+  protected readonly getDocId: GetDocId<T>;
+
+  constructor(getDocId: GetDocId<T>) {
+    this.getDocId = getDocId;
+  }
+
+  get(): Observable<T[]> {
+    return of(this.docs);
+  }
+  upsert(docs: T[]): Observable<void> {
```

Comment:
What about using a `Set` as the class property to reduce the complexity of the code here and in `delete`?

Reply:
`Set` would make the code slightly simpler, but would constrain the user to using the same objects (`===`). Not really a concern for originally intended use case, but current implementation is more resilient.

Reply:
It's added context for implementing the interface